kpis/kpi_calculator.py

```diff
@@ -345,6 +358,53 @@ def get_energy(self, plot=False, plot_by_source=False):
 
         return self.ener_tot
 
+    def get_peak_electricity(self):
+        '''This method returns the measure of the total
+        peak 15-minute electricity demand in kW/m^2.
+
+        Parameters
+        ----------
+        None
+
+        Returns
+        -------
+        pele_tot: float
+            peak 15-minute electricity demand in kW/m^2.
+            Returns None if no electrical power used in model.
+
+        '''
+
+        # If no electricity in model return None, otherwise calculate
+        if len(self.sources_pele)==0:
+            self.pele_tot = None
+            self.pele_dict = None
+        else:
+            tim_data = np.array(self._get_data_from_last_index('time',self.i_last_pele))
+            df_pow_data_all = pd.DataFrame(index=tim_data)
+            # Calculate peak electricity
+            # [returns KW/m^2 - assumes power measured in Watts]
+            for source in self.sources_pele:
+                for signal in self.case.kpi_json[source]:
+                    pow_data = np.array(self._get_data_from_last_index(signal,self.i_last_pele))
+                    df_pow_data = pd.DataFrame(index=tim_data, data=pow_data, columns=[signal])
+                    df_pow_data_all = pd.concat([df_pow_data_all, df_pow_data], axis=1)
+            df_pow_data_all.index = pd.TimedeltaIndex(df_pow_data_all.index, unit='s')
+            df_pow_data_all = df_pow_data_all.resample('15T').mean()/self.case._get_area()/1000.
```

Comment:
Are we sure we want to take the mean value here? Taking the mean may average out the peak values but I think we're interested in the instantaneous peak. If so, interpolation may be required.

Reply:
Let's discuss on the issues page (https://github.com/ibpsa/project1-boptest/issues/388).